src/main/scala/net/snowflake/spark/snowflake/io/SnowflakeResultSetRDD.scala

```diff
@@ -125,9 +127,19 @@ case class ResultIterator[T: ClassTag](
     }
   }
 
+  context.addTaskCompletionListener[Unit](_ => closeResultSet())
+
   val isIR: Boolean = isInternalRow[T]
   val mapper: ObjectMapper = new ObjectMapper()
   var currentRowNotConsumedYet: Boolean = false
+  var resultSetIsClosed: Boolean = false
```

Comment:
It will be better to declare `var resultSetIsClosed: Boolean = false` before `addTaskCompletionListener()`.

Reply:
I moved the `addTaskCompletionListener` down a couple of lines.